src/Plugin/formatter/FormatterJsonApi.php

```diff
@@ -738,4 +738,36 @@ protected static function retrieveIncludedItem($type, $id, array $included) {
     return NULL;
   }
 
+  /**
+   * Builds the relationship links.
+   *
+   * @param string $parent_id
+   *   The ID of the host entity.
+   * @param string $public_field_name
+   *   The name of the relationship.
+   * @param ResourceInterface $resource_plugin
+   *   The resource plugin.
+   * @param string $identifier
+   *   The ID of the target entity on the relationship.
+   *
+   * @return array
+   *   The links.
+   */
+  protected function buildRelationshipLinks($parent_id, $public_field_name, ResourceInterface $resource_plugin, $identifier) {
+    // If there is a resource plugin for the parent, set the related
+    // links.
+    $links = array();
+    if ($resource = $this->getResource()) {
+      $links['related'] = $resource->versionedUrl('', array(
+        'absolute' => TRUE,
+        'query' => array(
+          'filter' => array($public_field_name => $identifier),
+        ),
+      ));
+      $links['self'] = $resource_plugin->versionedUrl($parent_id . '/relationships/' . $public_field_name);
+      return $links;
```

Comment:
You are right. PHPStorm refactor feature is very smart, but needs human 👀 on it.